contracts/components/scanners/ScannerRegistryEnable.sol

```diff
@@ -96,43 +96,9 @@ abstract contract ScannerRegistryEnable is ScannerRegistryManaged {
      * @param enable true for enabling, false for disabling
      */
     function _enable(uint256 scannerId, Permission permission, bool enable) internal {
-        _beforeScannerEnable(scannerId, permission, enable);
-        _scannerEnable(scannerId, permission, enable);
-        _afterScannerEnable(scannerId, permission, enable);
-    }
-
-
-    /**
-     * @notice Hook _before scanner enable
-     * @dev does nothing in this contract
-     * @param scannerId ERC721 token id of the scanner.
-     * @param permission the sender claims to have to enable the agent.
-     * @param value true if enabling, false if disabling.
-     */
-    function _beforeScannerEnable(uint256 scannerId, Permission permission, bool value) internal virtual {
-    }
-
-    /**
-     * @notice Logic for enabling or disabling the scanner.
-     * @dev sets the corresponding byte in _disabled bitmap for scannerId. Emits ScannerEnabled event.
-     * @param scannerId ERC721 token id of the scanner.
-     * @param permission the sender claims to have to enable the agent.
-     * @param value true if enabling, false if disabling.
-     */
-    function _scannerEnable(uint256 scannerId, Permission permission, bool value) internal virtual {
-        _disabled[scannerId].setTo(uint8(permission), !value);
-        emit ScannerEnabled(scannerId, isEnabled(scannerId), permission, value);
-    }
-
-    /**
-     * @notice Hook _after scanner enable
-     * @dev emits Router hook.
-     * @param scannerId ERC721 token id of the scanner.
-     * @param permission the sender claims to have to enable the agent.
-     * @param value true if enabling, false if disabling.
-     */
-    function _afterScannerEnable(uint256 scannerId, Permission permission, bool value) internal virtual {
-        
+        if (!isRegistered(scannerId)) revert ScannerNotRegistered(address(uint160(scannerId)));
+        _disabled[scannerId].setTo(uint8(permission), !enable);
+        emit ScannerEnabled(scannerId, isEnabled(scannerId), permission, enable);
```

Comment:
Is the idea that users would disable their scanners to opt out of migration and re-enable them to continue running them after the migration? It seems odd to me. We're removing most of the ability to change things about a scanner (e.g. metadata) yet we allow enabling them.

Reply:
1. Users can opt out of the migration by disabling
2. Until `_migrationEnd` , those scanners can get rewarded. There is not much to do with scanners from the contract realy. They cannot update metadata, but that field is actually not used (dev team asked for it, ended up not using it, that is going to happen no more, also I'm enabling setters for the actual ERC721 metadata to be used via `tokenURI` in a future PR)
The thing is, no more new Scanners and no more data that could create edge cases.
3. They are free to self migrate during the allowed period (and after, really), the old scanners will stop being rewarded at some point.

Reply:
So my understanding is:
- The automatic migration will happen at some date and time `T` over a period of time `e`.
- A node runner can opt out of the automatic migration by disabling their scanner node before `T` and re-enabling it after `T+e`.

This seems to imply that during the migration period `[T, T+e]` all scanning will be essentially paused? Is that right?

If not, I find it weird that a node runner would be forced to pause their own scanners just for the side effect of opting out of migration. Independently of that, it's also weird that opting out requires two on-chain actions instead of just one. Have we considered adding a new flag that users can enable to opt-out without having disabled? All that said, I think the way we're doing things now is acceptable, if only a little weird.